qiling/extensions/r2/r2.py

```diff
@@ -111,25 +115,33 @@ def __lt__(self, other):
         return self.fromaddr < other.fromaddr
 
 class R2:
-    def __init__(self, ql: Qiling, baseaddr=(1 << 64) - 1, loadaddr=0):
+    def __init__(self, target: Union[Qiling, bytes, str], baseaddr=(1 << 64) - 1, loadaddr=0, **kwargs):
         super().__init__()
-        self.ql = ql
         self.baseaddr = baseaddr  # r2 -B [baddr]   set base address for PIE binaries
         self.loadaddr = loadaddr  # r2 -m [addr]    map file at given address
         self.analyzed = False
         self._r2c = libr.r_core.r_core_new()
-        if ql.code:
-            self._setup_code(ql.code)
+        if type(target).__name__ == 'Qiling':  # avoid circular import
+            self.ql = target
+            code, path = target.code, target.path
+            self._setup_code(code) if code else self._setup_file(path)
+        elif isinstance(target, bytes):
+            self._setup_code(target)
+        elif isinstance(target, str):
+            self._setup_file(target)
         else:
-            self._setup_file(ql.path)
+            raise ValueError(f"R2 only accept Qiling/str/bytes, not {type(target)}")
 
     def _setup_code(self, code: bytes):
         path = f'malloc://{len(code)}'.encode()
         fh = libr.r_core.r_core_file_open(self._r2c, path, UC_PROT_ALL, self.loadaddr)
         libr.r_core.r_core_bin_load(self._r2c, path, self.baseaddr)
         self._cmd(f'wx {code.hex()}')
         # set architecture and bits for r2 asm
-        self._cmd(f"e,asm.arch={self.ql.arch.type.name.lower().removesuffix('64')},asm.bits={self.ql.arch.bits}")
+        try:
+            self._cmd(f"e,asm.arch={self.ql.arch.type.name.lower().removesuffix('64')},asm.bits={self.ql.arch.bits}")
```

Comment:
`removesuffix` was introduced in Python 3.9 and we officially support 3.8.
That said, I think this is a hacky and instable way to map Qiling arch names to Radare2 arch names.